README.md

```diff
@@ -0,0 +1,265 @@
+- [ ] Library should be able to stream the output of a running job.
+  - [ ] Discovering new output should be efficient, avoid busy-waiting or
+        polling.
+  - [ ] Output should be from start of process execution.
+  - [ ] Multiple concurrent clients should be supported.
+  - [ ] Do not make any assumptions about the process's output - it may be text
+        or raw binary data.
+
+### API
+
+- [ ] [GRPC](https://grpc.io) API to start/stop/get status/stream output of a running process.
+- [ ] Use mTLS authentication and verify client certificate. Set up strong set
+      of cipher suites for TLS and good crypto setup for certificates. Do not
+      use any other authentication protocols on top of mTLS.
+- [ ] Use a simple authorization scheme.
+
+### Client
+
+- [ ] CLI should be able to connect to worker service and start, stop, get
+      status, and stream output of a job.
+
+## Repository Organization & Conventions
+
+The repository follows
+[standard Go conventions on module organization](https://go.dev/doc/modules/layout).
+
+```sh
+tp-worker/
+  go.mod
+  go.sum
+  cmd/
+    tp-worker-server/
+      tp-worker-server.go
+    tp-worker-client/
+      tp-worker-client.go
+  internal/
+    authn/
+      authn.go
+      authn_test.go
+    authz/
+      authz.go
+      authz_test.go
+    job/
+      job.go
+      job_test.go
+    output-storage/
+      output-storage.go
+      output-storage_test.go
+  job/
+    job.go
+    job_test.go
+  proto/
+    tp-worker.proto
+```
+
+As much as possible, following [Effective Go](https://go.dev/doc/effective_go)
+coding conventions will be adhered to. This should help in avoiding some common
+pitfalls such as data races, crashes, concurrency issues, deadlocks, etc.
+
+Testing will focus primarily on happy path, with added tests for intentional
+error conditions, such as when a command is not found, output is not found, user
+doesn't have authorization, etc. Extensive mocking, table-driven tests, fuzzing,
+and mutation testing could be added later as unintentional error conditions are
+discovered, but are not within scope at this time.
+
+Since an adequate starting point for gRPC with mTLS is provided as an example in
+the
+[`grpc-go` repository,](https://github.com/grpc/grpc-go/tree/master/examples/features/encryption/mTLS)
+that will be used as an initial base for the server and client implementations
+to grow out of.
+
+## Usage Examples
+
+### Environment Variables
+
+#### Server
+```sh
+TP_WORKER_IP=127.0.0.1                          # --ip-addr
+TP_WORKER_PORT=8080                             # --port
+TP_WORKER_USER_CERTDIR=/path/to/cert/dir/       # --user-certs
+TP_WORKER_PRIVATE_KEY=/path/to/private/key.pem  # --private-key
+```
+
+#### Client
+```sh
+TP_WORKER_IP=127.0.0.1                                # --ip-addr
+TP_WORKER_PORT=8080                                   # --port
+TP_WORKER_CLIENT_PRIVATE_KEY=/path/to/private/key.pem # --private-key
```

Comment:
You'll need the client certificate that matches this private key, right?

Reply:
This is probably related to my confusion of mTLS with something else, above.